app/Http/Controllers/DashboardAdminController.php

```diff
@@ -219,6 +219,23 @@ public function registerUser(): RedirectResponse
         // create the user
         $user = User::create($validated);
 
+        $given_file = Request::file('profile_image')[0]['file'];
+        //Check if it is a valid file and valid file
+        if ($given_file instanceof \Illuminate\Http\UploadedFile && $given_file->isValid()) {
+            $allowedExtensions = ['jpg', 'jpeg', 'png', 'gif'];
+            $originalExtension = strtolower($given_file->getClientOriginalExtension());
+
+            //naming for all files in s3 Buckt: name nachname Studiengang
+            if (in_array($originalExtension, $allowedExtensions)) {
+                $filename = $user->firstname.$user->lastname.$user->course_id.'.'.$originalExtension;
```

Comment:
Would suggest to add underscores between firstname, lastname and course_id. 

Because firstname, lastname and course_id are not primary key there can be double entries. Maybe it's better to use the email address?

Reply:
We should use a uuid for this.

something like this should work
```php
use Illuminate\Support\Str;
$uuid = Str::uuid()->toString();
```